### PR TITLE
xenvm: drivers: xen: add new Xen-related features (event channels, irq console and consoleio Dom0 console)

### PR DESCRIPTION
**Pull request by firscity**
This pull-request introduces new features, related to Xen virtualization support.

It includes:
1. Xen enlighten driver. This driver allows usage of pre-defined Xen event-channels (XenBus, PV console) on single (boot) CPU, allocation and other VCPU registration (VCPUOP_register_vcpu_info) will be added in future versions. 
2. Interrupt-driven API for Xen PV console over event channels.
3. Consoleio driver for Zephyr as Dom0 (privileged) domain (only for Dom0 console input/output work in regular Xen build, for debug build output works for other domain - already implemented in uart-xen-hvc).

Also XenVM doc was updated according to new changes.

### CODE REVIEW

**Comment by povergoing** [comment]
Just out of curiosity, how to create a DomU when Zephyr acts as Dom0? Would you like to describe the instructions in the documents?

File: drivers/xen/events.c [hunk 1]
```diff
@@ -9,17 +9,178 @@
...
+			events_pending &= (1 << event_index);
+
+			port = (pos_index * 8 * sizeof(xen_ulong_t))
+					+ event_index;
+			process_event(port);
+		}
+	}
+}
+
+
```
[the remark this answers is not in the document]
Comment (firscity) [reply drivers/xen/events.c]: thank you, fixed.

**Comment by firscity** [comment]
> Just out of curiosity, how to create a DomU when Zephyr acts as Dom0?

This is huge feature I am currently working on, but it stills in progress. So it is not ready to any kind of instructions or docs, sorry.
During development I just trying to upstream some helpful things to Zephyr mainline, Dom0 console is one of them.

**Comment by dcpleung** [comment]
Recommend changing to `hvc_xen_consoleio` for filenames, so that it will be next to `uart_hvc_xen.c` in directory listing. Same goes for kconfig by moving `CONSOLEIO` to the end.

**Comment by firscity** [comment]
> Recommend changing to `hvc_xen_consoleio` for filenames, so that it will be next to `uart_hvc_xen.c` in directory listing. Same goes for kconfig by moving `CONSOLEIO` to the end.

Thank you for great suggestion! Done.

File: CODEOWNERS [hunk 2]
```diff
@@ -345,6 +345,7 @@
 /drivers/serial/*esp32c3*                 @uLipe
 /drivers/serial/*esp32s2*                 @glaubermaroto
 /drivers/serial/Kconfig.xen               @lorc @firscity
+/drivers/serial/uart_consoleio_hvc_xen.c  @lorc @firscity
```
[the remark this answers is not in the document]
Comment (firscity) [reply CODEOWNERS]: Oh, sorry, forget about it. Thanks, done.

**Review by dcpleung — APPROVED** [review]

**Review by povergoing — APPROVED** [review]

**Comment by firscity** [comment]
Dear @carlocaione, can you please take a look on this PR?

**Review by carlocaione — APPROVED** [review]
This is ok but you need to fix the conflicting file issue.

**Comment by firscity** [comment]
Looks like some CI tests were stuck. I decided to re-pushed commits, no changes were made by last force-push.

**Comment by firscity** [comment]
The only one test that failing is:
```
-:640: WARNING:NEW_TYPEDEFS: do not add new typedefs
#640: FILE: include/xen/events.h:20:
+typedef struct event_channel_handle evtchn_handle_t;

- total: 0 errors, 1 warnings, 582 lines checked
```

I think it is okay and PR can be merged.

**Comment by carlocaione** [comment]
@nashif we need your superpower here

**Comment by firscity** [comment]
Dear @nashif, can you please process this?

**Comment by klogg** [comment]
Hi @carlocaione @nashif, are there any issues with this PR that block it from merging?

**Pull request merged**